### PR TITLE
fix(#2805): find-phase returns null phase_dir for archived phases (not archive path)

### LINKED ISSUE #2805: init.plan-phase / init.execute-phase return archived prior-milestone directory when current-milestone phase has no dir yet
## Summary

`gsd-sdk query init.plan-phase <N>` (and `init.execute-phase <N>`) returns the directory of an archived prior-milestone phase when the current milestone has a phase with the same number but no directory yet. The query returns a mixed payload: `phase_dir` from the archive, `phase_req_ids` from the current `ROADMAP.md`. Downstream `/gsd-plan-phase` and `/gsd-execute-phase` workflows then operate against the archive — corrupting completed-milestone history and skipping the actual unstarted current phase.

## Reproduction

1. Project has shipped milestone `vX` with phases archived under `.planning/milestones/vX-phases/02-something/`.
2. Start milestone `vX+1` with a Phase 2 in `.planning/ROADMAP.md`, but do not create `.planning/phases/02-*/` yet.
3. Run:

\`\`\`bash
gsd-sdk query init.plan-phase 2
\`\`\`

Observed:

\`\`\`json
{
  "phase_found": true,
  "phase_dir": ".planning/milestones/vX-phases/02-something",
  "phase_number": "02",
  "phase_name": "something",
  "phase_req_ids": "FND-07, FND-09 ..."
}
\`\`\`

Expected:

\`\`\`json
{
  "phase_found": true,
  "phase_dir": null,
  "phase_number": "2",
  "phase_name": "<current-roadmap phase name>",
  "phase_slug": "<derived from current ROADMAP>",
  "phase_req_ids": "<from current ROADMAP>"
}
\`\`\`

## Root Cause

\`findPhase\` in \`dist/query/phase.js\` searches \`.planning/phases/\` first, then falls through to archived \`.planning/milestones/v*-phases/\` (newest first) on a miss. When the archive matches it returns the archived directory with \`archived: <version>\` set.

\`getPhaseInfoWithFallback\` in \`dist/query/init.js\` only consults \`ROADMAP.md\` as a fallback when \`findPhase\` returns \`found: false\`. It never inspects the \`archived\` flag, so a stale archived hit suppresses the ROADMAP fallback even though the current ROADMAP has the phase.

## Blast Radius

Every milestone-2+ project where a current-milestone phase has the same number as an archived phase is affected — i.e., almost every long-running project after its first milestone. The risk window opens whenever a current phase has no directory yet (i.e., immediately after roadmap creation, before \`/gsd-plan-phase\`).

Affected handlers (all use \`getPhaseInfoWithFallback\`):
- \`init.plan-phase\` (\`dist/query/init.js:240\`)
- \`init.execute-phase\` (\`dist/query/init.js:170\`)
- handler at \`dist/query/init.js:440\`

## Suggested Fix

In \`getPhaseInfoWithFallback\`, treat an \`archived\` hit as not-yet-created when the phase exists in the current ROADMAP:

\`\`\`js
const archivedHitMasksCurrent = phaseInfo?.found && phaseInfo.archived && roadmapPhase?.found;
if ((!phaseInfo || !phaseInfo.found || archivedHitMasksCurrent) && roadmapPhase?.found) {
    const phaseName = roadmapPhase.phase_name;
    phaseInfo = {
        found: true,
        directory: null,
        phase_number: roadmapPhase.phase_number,
        phase_name: phaseName,
        phase_slug: phaseName ? generateSlugInternal(phaseName) : null,
        plans: [],
        summaries: [],
        incomplete_plans: [],
        has_research: false,
        has_context: false,
        has_verification: false,
        has_reviews: false,
    };
}
\`\`\`

This preserves the archived fallback for read-only queries against phases that no longer exist in the current ROADMAP, while preferring the current ROADMAP for ambiguous cases.

## Environment

- \`@gsd-build/sdk\` v0.1.0
- Node v24.14.0
- Windows 11 (also reproducible on macOS/Linux — the bug is in path-agnostic JS logic)

## Workaround

Apply the patch above to \`dist/query/init.js\` locally, or pre-create the current-milestone phase directory before running \`init.plan-phase\`/\`init.execute-phase\` so \`findPhase\` resolves the current dir first.

### PR DESCRIPTION
**Pull request by trek-e**
## What

When looking up a completed/archived phase, `gsd-sdk query find.phase <n>` returned the archive directory path as `phase_dir`. Workflows checking `phase_dir` to locate working files would then read from or write to the archive instead of treating the phase as inactive.

## Why

The `findPhase` function returned the archive path when a phase was found only in the archive, exposing a path that should not be used for active workflow operations.

## How

- Code fix was already applied; this PR adds the regression test `tests/bug-2805-archived-phase-fallback.test.cjs` to prevent future drift
- Test asserts: `phase_dir` is `null` for archived phases, `phase_found` is `true`, `phase_name` resolves from ROADMAP not archive

Closes #2805

## Release Notes

* **Bug Fixes**
  * Enhanced Codex hooks migrator with five edge-case corrections addressing robustness issues during hooks schema path migration and preventing invalid table creation.

* **Tests**
  * Added regression test for phase resolution correctness in archived milestone scenarios.

* **Documentation**
  * Updated changelog and v1.39.0-rc.5 release notes documenting migrator improvements.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -42,6 +42,19 @@ Format follows [Keep a Changelog](https://keepachangelog.com/en/1.1.0/).
...
+  regex, so hyphenated keys (e.g. `status-message`) and quoted keys are no longer
+  silently dropped; (2) `buildNestedBlock` no longer synthesises an empty
+  `[[hooks.TYPE.hooks]]` sub-table for matcher-only sections that carry no handler
+  fields — previously produced a broken entry with `type = "command"` but no
+  `command`; (3) the `legacyMapSections` filter now uses the parsed segment count
+  instead of dot-splitting the path string, preventing three-segment tables such as
+  `[hooks.SessionStart.hooks]` from being misclassified as event entries (same class
+  of bug fixed for `staleNamespacedAotSections` in the previous round); (4) regression
+  test added: `[[hooks."before.tool"]]` (a quoted key containing a dot) is correctly
+  treated as a two-segment namespace and not split on the inner dot. (#2809)
```
[the remark this answers is not in the document]
Comment (trek-e) [reply CHANGELOG.md:57]: Deferring to PR #2813 to avoid merge conflicts — that is the first PR touching this shared CHANGELOG line and it already applies the fix (five → four). Will be resolved when PR #2813 merges first.
